doc/pmon/pmon-chassis-requirements.md

```diff
@@ -0,0 +1,37 @@
+Section 1 Requirements that are needed by default:-
+        1. On LC the reboot command should power-cycle the entire LC . Expectation is Peer node should detect link down when reboot is given on LC
+		2. On RP the reboot command should reboot the entire system (RP and LC). . Expectation is Peer node should detect link down when reboot is given on RP
```

Comment:
Graceful reboot for LC vs power down for LC ? Open question.

Reply:
> if Line-card can not detect sup going down then sup after comin-up broadcast all LC's to self-reboot

This might have an issue! There are chassis/platform whereby LC comes up on their own the moment power is turned-on to the chassis. So, Supervisor asking for its boot-up may interrupt already booting-up LC in a chassis power-on/reload scenarios. Suggest to have Supervisor 'booting up' workflow to be same regardless of SUP-only reload/boot-up or chassis reload.

Reply:
Summary based of internal discussion:
1. If Supervisor goes down unplanned/ungracefully then there is no need to reboot LC or any other action.
2. In above scenario LC should be generating syslog complaining about Supervisor not being reachable (Eg: PMON trying to access Chassis DB to push the data)
3. Above syslog can be used by Alerting logic and necessary action can be taken form LC/Chassis perspective like doing `config reload` on LC.

Reply:
Regarding (1),  "there is no need to reboot LC or any other action"
There are a number of concerns with letting linecards run headless while a supervisor reboots
- There will be a long window of time where no state exchange between linecards will be possible due to lack of chassis_db. 
- The software on the linecard now has to deal with live disconnection/reconnection of connectivity  to chassis_db or other supervisor
- There's no real stated benefit motivating this propsal. A strong working assumption behind the chassis architecture conception for the current set of use cases is that there is enough redundancy in the network to easily tolerate the entire chassis going down and coming back, especially when it is a rare event like an unplanned supervisor rebbot.
- 
 
In general, it is best to keep failure handling simple and predictable, and avoid divergent flows across different scenarios. So unless there is a specific concrete problem that is solved by keeping the linecards, it would be strongly preferable to always reboot linecards when a supervisor reboots.

At the very least, the headless operation should be made optional and not mandated.

Reply:
Option 1 (Preferable)
If Supervisor goes down (headless operation) then LC should go down

Option 2
IF LC are running in headless mode the LC should be able to send syslog asap (mid-plane connectivity should get restored) so that LC can communicate the error to External Management.